versioned/persist/mongodb/src/test/java/org/projectnessie/versioned/persist/mongodb/FlapdoodleMongoTestConnectionProviderSource.java

```diff
@@ -120,4 +80,10 @@ public void stop() throws Exception {
       }
     }
   }
+
+  private Integer findRandomOpenPortOnAllLocalInterfaces() throws IOException {
+    try (ServerSocket socket = new ServerSocket(0)) {
+      return socket.getLocalPort();
```

Comment:
Technically this does not guarantee that Mongo will be able to bind to this port later. Is monitoring Mongo logs that much of an inconvenience?

Reply:
monitoring the mongo log depends on the logging format :shrug:

Reply:
true, but it's a solid test failure when it changes vs. a random failure with port allocation in odd environments... :shrug: 

@snazy : What is your take of this? I believe we talked briefly about this before :)

Reply:
alternatively I was thinking whether there's a way to ask flapdoodle for the port that mongo is running on

Reply:
May a PR to Flapdoodle? ;-)

Reply:
There's no way to let mongod tell you in a machine readable way which port it has chosen.

But Unix domain sockets help here #2863